examples/kotlin/src/main/kotlin/com/example/booktest/postgresql/QueriesImpl.kt

```diff
@@ -0,0 +1,293 @@
+// Code generated by sqlc. DO NOT EDIT.
+
+package com.example.booktest.postgresql
+
+import java.sql.Connection
+import java.sql.SQLException
+import java.sql.Types
+import java.time.OffsetDateTime
+
+const val booksByTags = """-- name: booksByTags :many
+SELECT 
+  book_id,
+  title,
+  name,
+  isbn,
+  tags
+FROM books
+LEFT JOIN authors ON books.author_id = authors.author_id
+WHERE tags && ?::varchar[]
+"""
+
+data class BooksByTagsRow (
+  val bookId: Int,
+  val title: String,
+  val name: String,
+  val isbn: String,
+  val tags: List<String>
+)
+
+const val booksByTitleYear = """-- name: booksByTitleYear :many
+SELECT book_id, author_id, isbn, booktype, title, year, available, tags FROM books
+WHERE title = ? AND year = ?
+"""
+
+data class BooksByTitleYearParams (
+  val title: String,
+  val year: Int
+)
+
+const val createAuthor = """-- name: createAuthor :one
+INSERT INTO authors (name) VALUES (?)
+RETURNING author_id, name
+"""
+
+const val createBook = """-- name: createBook :one
+INSERT INTO books (
+    author_id,
+    isbn,
+    booktype,
+    title,
+    year,
+    available,
+    tags
+) VALUES (
+    ?,
+    ?,
+    ?,
+    ?,
+    ?,
+    ?,
+    ?
+)
+RETURNING book_id, author_id, isbn, booktype, title, year, available, tags
+"""
+
+data class CreateBookParams (
+  val authorId: Int,
+  val isbn: String,
+  val booktype: BookType,
+  val title: String,
+  val year: Int,
+  val available: OffsetDateTime,
+  val tags: List<String>
+)
+
+const val deleteBook = """-- name: deleteBook :exec
+DELETE FROM books
+WHERE book_id = ?
+"""
+
+const val getAuthor = """-- name: getAuthor :one
+SELECT author_id, name FROM authors
+WHERE author_id = ?
+"""
+
+const val getBook = """-- name: getBook :one
+SELECT book_id, author_id, isbn, booktype, title, year, available, tags FROM books
+WHERE book_id = ?
+"""
+
+const val updateBook = """-- name: updateBook :exec
+UPDATE books
+SET title = ?, tags = ?
+WHERE book_id = ?
+"""
+
+data class UpdateBookParams (
+  val title: String,
+  val tags: List<String>,
+  val bookId: Int
+)
+
+const val updateBookISBN = """-- name: updateBookISBN :exec
+UPDATE books
+SET title = ?, tags = ?, isbn = ?
+WHERE book_id = ?
+"""
+
+data class UpdateBookISBNParams (
+  val title: String,
+  val tags: List<String>,
+  val isbn: String,
+  val bookId: Int
+)
+
+class QueriesImpl(private val conn: Connection) {
+
+  @Throws(SQLException::class)
```

Comment:
what does this do? force java code to catch `SQLException`?

Reply:
Yeah, basically all JDBC functions throw `SQLException`, so I figured it's nice to annotate it, and for it to be checked in Java